### PR TITLE
Drop old versions of sf

### PR DESCRIPTION
**Pull request by core23**
I am targeting this branch, because this is BC.

## Changelog

No  changelog needed
<!--
```markdown
### Removed
- Support for old versions of Symfony.
```
>

### CODE REVIEW

File: Form/Type/Filter/ChoiceType.php [hunk 1]
```diff
@@ -119,9 +120,7 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setDefaults([
             'field_type' => 'choice',
             'field_options' => [],
-            'operator_type' => method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')
-                ? 'Symfony\Component\Form\Extension\Core\Type\ChoiceType'
-                : 'choice', // NEXT_MAJOR: Remove ternary (when requirement of Symfony is >= 2.8)
+            'operator_type' => self::class,
```
Comment (jordisala1991) [inline Form/Type/Filter/ChoiceType.php]: This seems wrong

**Comment by jordisala1991** [comment]
If you only want to change form types to FQCNs can you make your changelog more specific? in this case there are other things that need to be dropped. Like the old code for Symfony Commands

**Comment by greg0ire** [comment]
I think this could be considered pedantic. Technically, support for old sf versions has been dropped and you wouldn't want that to appear twice int the changelog IMO

**Comment by Soullivaneuh** [comment]
@greg0ire This is also why it should be done at same time as version dropping next time: https://github.com/sonata-project/dev-kit/pull/318

But yes, late make this pedantic as the version drop was already done before.

**Comment by jordisala1991** [comment]
Thanks @core23

**Comment by AnatolyMuntean** [comment]
Seems like this change-set introduced some sort of naming conflict:
`PHP Fatal error:  Cannot use Symfony\Component\Form\Extension\Core\Type\ChoiceType as ChoiceType because the name is already in use in /var/www/dev/sf-test/vendor/sonata-project/admin-bundle/src/Form/Type/Filter/DateType.php on line 15`

Affected file are the ones in the same namespace as the overridden `ChoiceType` class.

Symfony v3.3.31 and SonataAdminBundle v3.25.1

**Comment by greg0ire** [comment]
Reported multiple times.

**Pull request merged**